### PR TITLE
[Kubernetes][autoscaler] Deep copy in K8s Node Provider to fix scaling issues

### LINKED ISSUE #14400: [k8s] min_workers doesn't seem to be honored
### What is the problem? & Reproduction (REQUIRED)

Ray version 1.2, k8s 18.6 and 19.x, Ubuntu 18.04

i’ve been playing around with various min_worker settings and have been seeing inconsistent behavior, e.g when i set min_worker to 10, the system indeed ended up with 10 pods, but when i changed it to 30, I only ended up with one or no worker.
The cluster yaml i used as a starting point is just this one: https://github.com/ray-project/ray/blob/6af02913470fa9f394e0fe37b8a7115e10a29a2b/python/ray/autoscaler/kubernetes/example-full.yaml#L98
.

To reproduce this, i think it should be sufficient to setup a larger k8s cluster and  run `ray up` with the modified cluster yaml i described above.
see this thread for further background information: https://discuss.ray.io/t/min-workers-doesnt-seem-to-be-honored/1002

 @richardliaw ^^

### PR DESCRIPTION
**Pull request by DmitriGekhtman**
## Why are these changes needed?

This PR fixes `KubernetesNodeProvider.create_node` by having it do a deep copy of its node config arg.

Currently the node provider does a shallow copy.
As a result, the config contents of the autoscaler's launch queue are mutated in high-load situations. This leads to the bizarre behavior reported in https://github.com/ray-project/ray/issues/14400 .

## Related issue number

~~This addresses https://github.com/ray-project/ray/issues/14400 when launching a large number of min_workers with the Ray operator.~~

~~With the cluster launcher (i.e. `ray up`) I got the correct number of worker pods but these didn't connect to Ray.~~

Closes https://github.com/ray-project/ray/issues/14400 .
(See discussion there for explanation of the strikeouts.)

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(
 
Due to our current testing set-up for the operator, I think a test validating the problem is solved for the operator should go in another PR.

### CODE REVIEW

**Comment by DmitriGekhtman** [comment]
^ need to fix my git push workflow to get rid of that messy commit history

**Review by edoakes — APPROVED** [review]
Oof, good catch!

File: python/ray/setup-dev.py [hunk 1]
```diff
@@ -75,6 +75,7 @@ def do_link(package, force=False, local_path=None):
     do_link("util", force=args.yes)
     do_link("serve", force=args.yes)
     do_link("_private", force=args.yes)
+    do_link("node.py", force=args.yes)
```
Comment (edoakes) [inline python/ray/setup-dev.py:78]: Is this intentional?

**Comment by edoakes** [comment]
@DmitriGekhtman to avoid the messy commit history:

- Never work off of your local `master`. Only pull `upstream/master` into it. When you work on a feature locally, do it in a local branch.
- Base all local branches off of local `master`. If you need to merge latest, do `git fetch upstream/master; git merge upstream/master`

When you start a new PR, you should basically always do: `git checkout master; git pull upstream master; git checkout -b new-feature-branch` unless you're intentionally working off another branch.

File: python/ray/setup-dev.py (hunk 1 above)
Comment (DmitriGekhtman) [reply python/ray/setup-dev.py:78]: Yes.
I've just started adding things to setup-dev when they cause me issues. 
(added _private in another PR)

I could remove this and add a PR later to link all of the python files in the root directory.

Comment (edoakes) [reply python/ray/setup-dev.py:78]: No worries, it seems fine to include it

Comment (DmitriGekhtman) [reply python/ray/setup-dev.py:78]: okidoke

**Comment by DmitriGekhtman** [comment]
I had somehow messed up my local master -- fixed with `git reset --hard upstream/master`

Aaanyways, I think this PR is good to merge.

**Pull request merged**